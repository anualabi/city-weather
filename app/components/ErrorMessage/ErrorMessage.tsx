import {
  StyleProp,
  StyleSheet,
  TextStyle,
  View,
  ViewStyle,
} from "react-native";

import Text from "@/components/Text";

export interface ErrorMessageProps {
  containerStyle?: StyleProp<ViewStyle>;
  message: string;
  textStyle?: StyleProp<TextStyle>;
}

export default function ErrorMessage({
  containerStyle,
  message,
  textStyle,
}: ErrorMessageProps) {
  if (!message) {
    return null;
  }

  return (
    <View style={[styles.container, containerStyle]} testID="errorContainer">
      <Text style={[styles.text, textStyle]}>{message}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    flex: 1,
    justifyContent: "center",
    padding: 10,
  },
  text: { color: "red", fontSize: 18, lineHeight: 28, textAlign: "center" },
});
