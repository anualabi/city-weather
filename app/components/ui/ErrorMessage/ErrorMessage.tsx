import {
  StyleProp,
  StyleSheet,
  TextStyle,
  View,
  ViewStyle,
} from "react-native";

import { colors } from "@/theme";
import Text from "@/components/ui/Text";

export interface ErrorMessageProps {
  containerStyle?: StyleProp<ViewStyle>;
  message: string;
  textStyle?: StyleProp<TextStyle>;
}

export default function ErrorMessage({
  containerStyle,
  message,
  textStyle,
}: ErrorMessageProps) {
  if (!message) {
    return null;
  }

  return (
    <View style={[styles.container, containerStyle]} testID="errorContainer">
      <Text style={[styles.text, textStyle]}>{message}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    flex: 1,
    justifyContent: "center",
    padding: 10,
  },
  text: {
    color: colors.error,
    fontSize: 18,
    lineHeight: 28,
    textAlign: "center",
  },
});
