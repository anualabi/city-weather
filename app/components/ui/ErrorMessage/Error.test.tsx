import "@testing-library/jest-native/extend-expect";
import { render } from "@testing-library/react-native";
import renderer from "react-test-renderer";

import ErrorMessage, { ErrorMessageProps } from "./ErrorMessage";

const customRender = (props: ErrorMessageProps) => {
  return render(<ErrorMessage {...props} />);
};

describe("ErrorMessage component", () => {
  test("should render correctly", () => {
    const tree = renderer
      .create(<ErrorMessage message="An error occurred" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should apply custom styles", () => {
    const { getByTestId } = customRender({
      message: "An error occurred",
      containerStyle: { backgroundColor: "grey" },
    });

    expect(getByTestId("errorContainer")).toHaveStyle({
      backgroundColor: "grey",
    });
  });
});
