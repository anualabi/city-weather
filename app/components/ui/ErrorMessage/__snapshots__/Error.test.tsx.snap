// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component should render correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 10,
      },
      undefined,
    ]
  }
  testID="errorContainer"
>
  <Text
    style={
      [
        {
          "color": "#343434",
          "fontSize": 16,
        },
        [
          {
            "color": "#C03403",
            "fontSize": 18,
            "lineHeight": 28,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    An error occurred
  </Text>
</View>
`;
